TopAnalysis/src/FCNCTriLeptonCppWorker.cc

```diff
@@ -126,6 +143,21 @@ bool FCNCTriLeptonCppWorker::isVetoElectron(const unsigned i) const {
 
   return true;
 }
+//nonprompt electrons
+bool FCNCTriLeptonCppWorker::isNPElectron(const unsigned i) const {
+  const double pt = in_Electrons_p4[0]->At(i);
+  const double eta = in_Electrons_p4[1]->At(i);
+  const double scEta = eta + in_Electrons_dEtaSC->At(i);
+  if ( pt < minElectronPt_ or std::abs(eta) > maxElectronEta_ ) return false;
+  if ( std::abs(scEta) > 1.4442 and std::abs(scEta) < 1.566 ) return false;
+  //nanoAOD object -> Electron_cutBased_Sum16 0:fail, 1:veto, 2:loose, 3:medium, 4:tight
+  if ( in_Electrons_id->At(i) != 2 ) return false;
```

Comment:
1. you are asking electron id == 2. this means you are rejecting medium, tight electrons.
2. need special electron id without isolation... no?